root/tag/DeleteTag.js

```diff
@@ -0,0 +1,71 @@
+/*
+ * @flow strict
+ * Copyright (C) 2021 MetaBrainz Foundation
+ *
+ * This file is part of MusicBrainz, the open internet music database,
+ * and is licensed under the GPL version 2, or (at your option) any
+ * later version: http://www.gnu.org/licenses/gpl-2.0.txt
+ */
+
+import * as React from 'react';
+
+import {CatalystContext} from '../context.mjs';
+import FormSubmit from '../static/scripts/edit/components/FormSubmit.js';
+
+import TagLayout from './TagLayout.js';
+
+type Props = {
+  +deleteDownvoted?: boolean,
+  +tag: TagT,
+};
+
+const DeleteTag = ({
+  deleteDownvoted = false,
+  tag,
+}: Props): React.Element<typeof TagLayout> | null => {
+  const title = deleteDownvoted
+    ? l('Delete tag downvotes')
+    : l('Delete tag uses (upvotes)');
+  const $c = React.useContext(CatalystContext);
+  const user = $c.user;
+
+  return user ? (
+    <TagLayout page="delete" tag={tag} title={title}>
+      <h2>{title}</h2>
+      <p>
+        {exp.l(
+          deleteDownvoted ? (
+            `Are you sure you want to remove
+             {tag_list|all your votes against the tag “{tag}”}?`
+          ) : (
+            `Are you sure you want to remove
+             {tag_list|all your uses (upvotes) of the tag “{tag}”}?`
+          ),
+          {
+            tag: tag.name,
+            tag_list: '/user/' + encodeURIComponent(user.name) +
+                      '/tag/' + encodeURIComponent(tag.name) +
+                      '?show_downvoted=' + (deleteDownvoted ? '1' : '0'),
+          },
+        )}
+      </p>
+      <p>
+        {l('This can take a long time if you delete a large amount of tags.')}
```

Comment:
Did you test the time it is taking or is it a guess?
Is there any way to know the amount of tags (actually tag votes) to be deleted?

Reply:
IIRC we did check the time and it's fairly slow (in the same way the site sometimes times out when deleting an account with many thousands of tags). But it was ages ago and I don't have the exact numbers anywhere.

Reply:
We have a way to know the amount of tags a person has (we show them on their profile page with other stats) so there's no way in theory why we couldn't do the same with a filter for the specific tag :)